independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/BootstrapAppModelResolver.java

```diff
@@ -264,6 +266,11 @@ public boolean visitLeave(DependencyNode node) {
 
         List<AppDependency> fullDeploymentDeps = new ArrayList<>(userDeps);
         fullDeploymentDeps.addAll(deploymentDeps);
+        //we need these to have a type of 'jar'
+        //type is blank when loaded
+        for (AppArtifactKey i : localProjects) {
+            appBuilder.addLocalProjectArtifact(new AppArtifactKey(i.getGroupId(), i.getArtifactId(), null, "jar"));
```

Comment:
Ideally, we should at least match this `localProjects` to the resolved user dependencies: first, to make sure they are actually present among the resolved deps; second, we can get the actual artifact classifier and types from them.